src/app/components/news/news.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit, Inject, Input } from '@angular/core';
+import { NewsService, ActualNewsService } from '../../services/news.service';
+import { News } from '../../models/News';
+
+@Component({
+  selector: 'app-news',
+  templateUrl: './news.component.html',
+  styleUrls: ['./news.component.scss'],
+  // Here we choose which news service is used
+  providers: [{ provide: 'NewsService', useClass: ActualNewsService }]
+})
+export class NewsComponent implements OnInit {
+  news: News[];
+
+  constructor(@Inject('NewsService') newsService: NewsService) {
```

Comment:
just out of curiosity, do you need to be explicit with the inject decorator here because you have the mock service?

Reply:
I'm using the `NewsService` interface in the constructor, but the provider is the class that implements it `ActualNewsService`. Angular can't resolve it by itself (presumably because class names are different), so I explicitly give the provider a name ('`NewsService'` in this case) and tell it to inject it there.

Reply:
I could be wrong but could you remove the providers line in the metadata and have just 
`@Inject(ActualNewsService) newService: NewsService` as the argument?

Also I had come across a [InjectableToken](https://angular.io/api/core/InjectionToken) when looking up other stuff which I think is what they intend for people to use instead of just a string (for 'NewService').